packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -2625,9 +2668,13 @@ export class SnapController extends BaseController<
   }
 
   #validateSnapPermissions(
-    processedPermissions: Record<string, Pick<PermissionConstraint, 'caveats'>>,
+    initialPermissions: Record<string, Pick<PermissionConstraint, 'caveats'>>,
+    dynamicPermissions?: Record<string, Pick<PermissionConstraint, 'caveats'>>,
   ) {
-    const permissionKeys = Object.keys(processedPermissions);
+    const permissionKeys = Object.keys({
+      ...initialPermissions,
```

Comment:
We should probably still require one handler permissions minimum in `initialPermissions`. Otherwise that Snap can't request its dynamic permissions 😅

Reply:
Updated.